File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n=nums.size();
        int s=0,m=0,e=n-1;
        while(m<=e){
            if(nums[m]==0){
                swap(nums[s],nums[m]);
                s++;
                m++;
            }else if(nums[m]==1){
                m++;
            }else{
                swap(nums[m],nums[e]);
                e--;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n=nums.length;
        int s=0,e=nums.length-1,i=0;
        while(i<=e){
            if(nums[i]==0){
               int temp=nums[i];
               nums[i]=nums[s];
               nums[s]=temp;
               s++;
               i++;
            }
            else if(nums[i]==1){
               i++;
            }
            else if(nums[i]==2){
               int temp=nums[i];
               nums[i]=nums[e];
               nums[e]=temp;
               e--;
            }
        }
    }
}